src/Microsoft.Health.Fhir.Core/Features/Search/SearchValues/StringSearchValue.cs

```diff
@@ -105,5 +105,10 @@ public override string ToString()
         {
             return String.EscapeSearchParameterValue();
         }
+
+        public override int GetHashCode()
+        {
+            return String != null ? String.GetHashCode(StringComparison.OrdinalIgnoreCase) : 0;
```

Comment:
>String != null ? [](http://example.com/codeflow?start=19&length=16)

nit:
I don't think you need this check.
Constructor of class make sure it's not null, and it's get only property, so it should never be null.

Reply:
Same I believe stands for DateTimeSearchValue.